core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.spec.ts

```diff
@@ -217,6 +227,25 @@ describe('WorkflowEditorComponent', () => {
       expect(jointHighlighterElementAfterUnhighlight.length).toEqual(0);
     });
 
+    it('should react to operator validation and change the color of operator box if the operator is valid ',
+         () => {
+    const jointGraphWrapper = workflowActionService.getJointGraphWrapper();
+    workflowActionService.addOperator(mockScanPredicate, mockPoint);
+    workflowActionService.addOperator(mockResultPredicate, mockPoint);
+    workflowActionService.addLink(mockScanResultLink);
+    const newProperty = { 'tableName': 'test-table' };
+    workflowActionService.setOperatorProperty(mockScanPredicate.operatorID, newProperty);
+    const operator1 = component.getJointPaper().getModelById(mockScanPredicate.operatorID);
+    const operator2 = component.getJointPaper().getModelById(mockResultPredicate.operatorID);
+    expect(operator1.attr('rect/stroke')).toEqual('green');
+    expect(operator2.attr('rect/stroke')).toEqual('green');
+
+
+  });
+
+
+
```

Comment:
Move this to `joint-ui.service.spec.ts`, read comments below.

Reply:
This test case is related to workflow action service and its main idea is to check whether operator can react to the validation logic. I think it's better written here.